src/cluster.c

```diff
@@ -3658,9 +3706,15 @@ void clusterCron(void) {
     while((de = dictNext(di)) != NULL) {
         clusterNode *node = dictGetVal(de);
         /* The protocol is that they return non-zero if the node was
-         * terminated. */
-        if(clusterNodeCronHandleReconnect(node, handshake_timeout, now)) continue;
+         * terminated.
+         * The sequence goes:
+         * 1. We try to shrink link buffers if possible.
+         * 2. We free the links whose buffers are still oversized after possible shrinking.
+         * 3. We immediately attempt reconnecting after freeing links.
+         */
         if(clusterNodeCronResizeBuffers(node)) continue;
+        if(clusterNodeCronFreeLinkOnBufferLimitReached(node)) continue;
```

Comment:
Not clear if these need to possibly continue, they don't ever kill the node.

Reply:
Yeah, but this function doesn't terminate nodes, it operates on links instead and never deletes the node.

Reply:
You are right, this function never deletes the node. So it's more of a question of why does it have a return value.